docs/topics/menu_system.rst

```diff
@@ -153,36 +153,38 @@ Many of the methods below pass this list of nodes to the ones it calls, and retu
-    * :py:meth:`menus.templatetags.menu_tags.ShowMenu.get_context()`
-        * :py:meth:`menus.menu_pool.MenuPool.get_nodes()`
-            * :py:meth:`menus.menu_pool.MenuPool.discover_menus()` checks every application's ``cms_menus.py``, and registers:
-                * Menu classes, placing them in the ``self.menus`` dict
-                * Modifier classes, placing them in the self.modifiers list
-            * :py:meth:`menus.menu_pool.MenuPool._build_nodes()`
-                * checks the cache to see if it should return cached nodes
-                * loops over the Menus in self.menus (note: by default the only generator is :py:class:`cms.menu.CMSMenu`); for each:
-                    * call its :py:meth:`menus.base.Menu.get_nodes()` - the menu generator
-                    * :py:func:`menus.menu_pool._build_nodes_inner_for_one_menu()`
-                    * adds all nodes into a big list
-            * :py:meth:`menus.menu_pool.MenuPool.apply_modifiers()`
-                * :py:meth:`menus.menu_pool.MenuPool._mark_selected()`
-                * loops over each node, comparing its URL with the request.path_info, and marks the best match as ``selected``
-                * loops over the Modifiers in ``self.modifiers`` calling each one's :py:meth:`~menus.base.Modifier.modify()` with ``post_cut=False``. The default Modifiers are:
-                    * :py:class:`cms.menu.NavExtender`
-                    * :py:class:`cms.menu.SoftRootCutter` removes all nodes below the appropriate soft root
-                    * :py:class:`menus.modifiers.Marker` loops over all nodes; finds selected, marks its ancestors, siblings and children
-                    * :py:class:`menus.modifiers.AuthVisibility` removes nodes that require authorisation to see
-                    * :py:class:`menus.modifiers.Level` loops over all nodes; for each one that is a root node (``level == 0``) passes it to:
-                        * :py:meth:`~menus.modifiers.Level.mark_levels()` recurses over a node's descendants marking their levels
-        * we're now back in :py:meth:`menus.templatetags.menu_tags.ShowMenu.get_context()` again
-        * if we have been provided a root_id, get rid of any nodes other than its descendants
-        * :py:meth:`menus.templatetags.menu_tags.cut_levels()` removes nodes from the menu according to the arguments provided by the template tag
-        * :py:meth:`menus.menu_pool.MenuPool.apply_modifiers()` with ``post_cut = True`` loops over all the Modifiers again
-            * :py:class:`cms.menu.NavExtender`
-            * :py:class:`cms.menu.SoftRootCutter`
-            * :py:class:`menus.modifiers.Marker`
-            * :py:class:`menus.modifiers.AuthVisibility`
-            * :py:class:`menus.modifiers.Level`:
-                * :py:meth:`menus.modifiers.Level.mark_levels()`
-        * return the nodes to the context in the variable ``children``
+.. |_| unicode:: 0xA0 0xA0 0xA0
```

Comment:
Not in favour of this solution, it would be better to find a way to reduce the nesting if necessary.

Reply:
Do you really want the nesting structure or just a tabular appearance? I have added dots to give a better look. I will try to show the result on readthedocs.

Reply:
Please, compare even the first nested levels here:
https://media.readthedocs.org/pdf/django-cms/latest/django-cms.pdf (p.110) with there:
https://media.readthedocs.org/pdf/patched-djangocms-doc/latest/patched-djangocms-doc.pdf (p.109)

Also, here:
http://docs.django-cms.org/en/latest/topics/menu_system.html and there:
http://patched-djangocms-doc.readthedocs.io/en/latest/topics/menu_system.html

I do agree, my solution is not perfect yet, by using nested levels will be never good I'm afraid.

Reply:
@fp4code I feel the nesting is necessary, because the levels of nesting represent the logic of the system. It's true that it doesn't look good in the PDF version, but on the other hand, far more people will use the HTML, which is more important.

What is the maximum level of nesting that the PDF renditions can handle? It may be possible to break apart the monolithic list into smaller, shallower ones.

Reply:
@fp4code Re the third issue (Japanese), I'm also happy with that.